include/xtensor-io/xio_blosc.hpp

```diff
@@ -16,15 +16,40 @@
 #include "xtensor-io.hpp"
 #include "blosc.h"
 
-bool blosc_initialized = false;
-
 namespace xt
 {
     namespace detail
     {
+        class xio_blosc
+        {
+        public:
+            static void initialize()
+            {
+                if (!instance().m_initialized)
+                {
+                    blosc_init();
+                    instance().m_initialized = true;
+                }
+            }
+
+        private:
+            xio_blosc()
+            {
+                m_initialized = false;
+            }
+
+            static xio_blosc& instance()
+            {
+                static xio_blosc instance;
+                return instance;
+            }
+            bool m_initialized;
+        };
```

Comment:
This is a bit of overkill if you don't need additional members in `xio_blosc`. This could be simplified into

```cpp
void init_blosc()
{
    static bool initialized = false;
    if (!initialized)
    {
        blosc_init();
        initialized = true;
    }
}
```

Reply:
Indeed!